### PR TITLE
RONDB-382: Improve builds and build information

### PR DESCRIPTION
**Pull request by mronstro**
Upgrade to using GCC 10 from GCC 9 (about 5% improvement)
Upgrade to using CMAKE_BUILD_TYPE=Release from RelWithDebinfo (-O2 to -O3) (about 5% improvement)
Changed a few things to RonDB in build information.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -279,12 +279,11 @@ IF(CMAKE_HOST_UNIX AND NOT FORCE_UNSUPPORTED_COMPILER
...
         SET(ALTERNATIVE_PATHS "/opt/rh/devtoolset-8")
       ELSEIF(LINUX_RHEL7)
-        # ARM requires GCC 10, but we still build releases for
-        # x86_64 on GCC 8.
+        # ARM requires GCC 10. GCC 10 is faster than GCC 9
         IF(MY_HOST_MACHINE_NAME MATCHES "aarch64")
           SET(ALTERNATIVE_PATHS "/opt/rh/devtoolset-10")
         ELSE()
-          SET(ALTERNATIVE_PATHS "/opt/rh/devtoolset-9")
+          SET(ALTERNATIVE_PATHS "/opt/rh/devtoolset-10")
```
Comment (olapiv) [inline CMakeLists.txt]: Do you also want to adjust the `update-alternatives` command in Dockerfile.ubuntu22?

File: Dockerfile.oraclelinux7 [hunk 2]
```diff
@@ -95,7 +95,7 @@ RUN --mount=type=cache,target=$DOWNLOADS_CACHE_DIR \
 # CMake will look for $OPENSSL_ROOT_DIR
 ENV OPENSSL_ROOT_DIR=$OPENSSL_ROOT
 RUN --mount=type=cache,target=$DOWNLOADS_CACHE_DIR \
-    source scl_source enable devtoolset-9 \
+    source scl_source enable devtoolset-10 \
```
Comment (olapiv) [inline Dockerfile.oraclelinux7]: Search for the comments such as `gcc/g++ 9/10` in the Dockerfile. Can you update them as well?

**Review by olapiv — CHANGES_REQUESTED** [review]

**Review by smkniazi — DISMISSED** [review]
Looks OK,

File: Dockerfile.oraclelinux7 (hunk 2 above)
Comment (mronstro) [reply Dockerfile.oraclelinux7]: ok, done

File: CMakeLists.txt (hunk 1 above)
Comment (mronstro) [reply CMakeLists.txt]: ok, done

**Review by olapiv — APPROVED** [review]

**Pull request merged**